app/models/garden.rb

```diff
@@ -12,8 +12,10 @@ def add_plant(plant)
     GardenPlant.create!(garden: self, plant: plant, planted_date: Date.today)
   end
 
-  def create_and_add_plant(plant_params)
-    plant = Plant.new(plant_params)
+  def create_and_add_plant(plant_id, plant_params)
+    plant = Plant.new(plant_params) do |plant|
+      plant.id = plant_id 
+    end
```

Comment:
Wonder if this will cause problems in the future. As long as user's don't have the ability to add a plant then we shouldn't run into duplicate key creation.